invoices/invoiceregistry.go

```diff
@@ -96,21 +99,33 @@ func NewRegistry(cdb *channeldb.DB, finalCltvRejectDelta int32) *InvoiceRegistry
 		hodlSubscriptions:         make(map[channeldb.CircuitKey]map[chan<- interface{}]struct{}),
 		hodlReverseSubscriptions:  make(map[chan<- interface{}]map[channeldb.CircuitKey]struct{}),
 		finalCltvRejectDelta:      finalCltvRejectDelta,
+		expiryWatcher:             expiryWatcher,
 		quit:                      make(chan struct{}),
 	}
 }
 
 // Start starts the registry and all goroutines it needs to carry out its task.
 func (i *InvoiceRegistry) Start() error {
-	i.wg.Add(1)
+	// Start InvoiceExpiryWatcher and prepopulate it with existing active
+	// invoices.
+	if err := i.expiryWatcher.Start(i.CancelInvoice); err != nil {
+		return err
+	}
 
+	if err := i.expiryWatcher.PrefetchInvoices(i.cdb); err != nil {
+		return err
+	}
```

Comment:
Could merge these calls, prefetch inside start. Unless we use them already anywhere separately.

Reply:
Initially I wanted the API to be more flexible, but now that it's finished I agree with you, there's no need to separate this. Done.

Reply:
So prefetch is now part of `InvoiceRegisty`. This way `InvoiceExpiryWatcher` won't depend on `channeldb.DB` (suggested by @Roasbeef)